light-system-programs/programs/merkle_tree_program/src/config_accounts/merkle_tree_authority.rs

```diff
@@ -42,6 +45,22 @@ pub struct InitializeMerkleTreeAuthority<'info> {
         space = 8880
     )]
     pub transaction_merkle_tree: AccountLoader<'info, TransactionMerkleTree>,
+    #[account(
+        init,
+        seeds = [
+            EVENT_MERKLE_TREE_SEED,
+            0u64.to_le_bytes().as_ref(),
+        ],
+        bump,
+        payer=authority,
+        // discriminator + height (u64) + filled subtrees ([[u8; 32]; 18]) +
+        // roots ([[u8; 32]; 20]) + next_index (u64) + current_root_index (u64)
+        // + hash_function (enum) + merkle_tree_nr (u64) + newest (u8) +
+        // padding (7 * u8)
+        // 8 + 8 + 18 * 32 + 20 * 32 + 8 + 8 + 8 + 8 + 8 + 1 + 7 = 1280
+        space = 1280,
```

Comment:
not directly related to this pr:
do you think we could have the macro calculate a default space value as long as there are no vectors defined in the struct?

Reply:
I was genuinely thinking about that - even about having that macro as a very generic one, which could be used in any Anchor project. Seems doable, I think it would be a matter of iterating over all fields and doing `std::mem::size_of::<TYPE>()` on all of them.

Reply:
let's put it on the list

Reply:
Done.

I think an another useful macro would be one which automatically adds padding fields and ensures alignment for structs.

Such a macro would be useful also for Aya, so probably worth to be done as a standalone crate.